authorization/src/main/kotlin/org/modelix/authorization/permissions/ModelServerSchema.kt

```diff
@@ -0,0 +1,142 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.authorization.permissions
+
+val modelServerSchema = buildPermissionSchema {
+    resource("model-server") {
+
+        permission("admin") {
+        }
+    }
+
+    resource("permission-schema") {
+        permission("write") {
+            includedIn("model-server", "admin")
+            permission("read")
+        }
+    }
+
+    resource("repository") {
+        parameter("name")
+
+        permission("admin") {
+            includedIn("model-server", "admin")
+            permission("rewrite") {
+                permission("delete")
+                permission("write") {
+                    permission("create")
+                    permission("read") {
+                        permission("list") {
+                            includedIn("branch", "list")
+                        }
+                    }
+                }
+            }
+        }
+
+        resource("objects") {
+            permission("read") {
+            }
+            permission("add") {
+                description("Can add new objects, but not modify existing ones")
+            }
+        }
+
+        resource("branch") {
+            parameter("name")
+
+            permission("admin") {
+                includedIn("repository", "admin")
+                permission("rewrite") {
+                    includedIn("repository", "rewrite")
+                    description("Destructive write operations that change the history and loses previously pushed changes.")
+
+                    permission("force-push") {
+                        description("Overwrite the current version. Don't do any merges and don't prevent losing history.")
+                    }
+                    permission("delete")
+                    permission("write") {
+                        description("Non-destructive write operations that preserve the history.")
+                        includedIn("repository", "write")
+                        permission("create") {
+                            description("Can create a branch with this name, if it doesn't exist yet.")
+                        }
+                        permission("push") {
+                            description("Add changes to a branch and merge it with the current version ")
+                        }
+                        permission("read") {
+                            permission("list") {
+                                includes("repository", "list")
+                                description("Allowed to know its existence and name, but not the content.")
+                            }
+                            permission("pull") {
+                                description("Allowed reading the version hash. Permissions on objects is checked on repository level, which mean if a client knows the hash it can still read the content.")
```

Comment:
```suggestion
                                description("Allowed reading the version hash. Permissions on objects are checked on repository level, which mean if a client knows the hash it can still read the content.")
```